test/functional/test_framework/authproxy.py

```diff
@@ -188,7 +188,12 @@ def _get_response(self):
                 {'code': -342, 'message': 'non-JSON HTTP response with \'%i %s\' from server' % (http_response.status, http_response.reason)},
                 http_response.status)
 
-        responsedata = http_response.read().decode('utf8')
+        data = http_response.read()
+        try:
+            responsedata = data.decode('utf8')
+        except UnicodeDecodeError as e:
+            raise JSONRPCException({
+                'code': -342, 'message': f'Cannot decode response in utf8 format, content: {data}, exception: {e}'})
```

Comment:
A nit but it's easier to read as well, catches eyes quickly.

```suggestion
                'code': -342, 'message': f'Cannot decode response in UTF-8 format, content: {data}, exception: {e}'})
```